app/javascript/packs/two_factor_authentication.js

```diff
@@ -29,8 +29,13 @@ function callback(url, body) {
   }).then(function(response) {
     window.location.replace(response.data.redirect_path);
   }).catch(function(error) {
-    window.location.reload();
-    console.error(error);
+    if (error.response.status === 422) {
+      const errorMessage = document.getElementById('security-key-error-message');
+      errorMessage.classList.remove('hidden');
+      console.error(error.response.data.error);
```

Comment:
This is a bit ugly, but what I'm trying to do is to display an error message to users when an invalid key is used – where an [`unprocessable_entity` response](https://github.com/tootsuite/mastodon/pull/14466/commits/d1a5b696bbadd4e06bf80651a41e07e8722692c3#diff-83f7f4d352a4996ad969d4824a551f9dL58) is returned –, but handling the error differently if it's caused by another reason.